### ISSUE TITLE
Default does not match AllowedPattern when Type is CommaDelimitedList

### ISSUE THREAD

**Issue by andyloughran** [aws-cloudformation/cfn-lint#2632]
### CloudFormation Lint Version

0.72.2

### What operating system are you using?

Mac

### Describe the bug

When running against a parameter with the Type CommaDelimitedList, E2015 incorrectly assumes that the Default (which is a list of values) should match AllowedPattern.

### Expected behavior

When Type is set to CommaDelimitedList, AllowedPattern should match each individual item in the list, not the entire Default string.

https://aws.amazon.com/about-aws/whats-new/2022/12/specify-validation-checks-commadelimitedlist-parameter-aws-cloudformation/

### Reproduction template

```yaml

Parameters:
 AWSOrganizationId:
    Description: "Comma-separated IDs of AWS Organizations."
    Type: "CommaDelimitedList"
    Default: "o-pfdf83rrys,o-35lsdsco28"
    AllowedPattern: "^[a-z0-9-]+$"
    ConstraintDescription: "must be a valid organization ID, made of lowercase letters and numbers"
    ```

**Comment by kddejong** [comment]
Fixed by #2521  and released.
